extensions/wikia/WikiaHubsServices/modules/WikiaHubsModuleService.class.php

```diff
@@ -10,25 +10,36 @@
 	private $shouldFilterCommercialData = false;
 
 	public function __construct($cityId) {
+		global $wgForeignFileRepos;
 		parent::__construct();
 
 		$this->cityId = $cityId;
 		$this->skinName = RequestContext::getMain()->getSkin()->getSkinName();
+
+		$hubWiki = WikiFactory::getWikiByID($cityId);
+		$hubDbName = $hubWiki->city_dbname;
+
+		$wgForeignFileRepos[] = [
+			'class'            => 'WikiaForeignDBViaLBRepo',
+			'name'             => $hubDbName,
+			'transformVia404'  => true,
+			'wiki'             => $hubDbName,
+			'backend'          => "wikia{$hubDbName}-backend"
+		];
```

Comment:
Did you check what happen if you add foreign repo to the wiki that you're currently on? eg. render hub page